nutanix/structure.go

```diff
@@ -177,73 +220,87 @@ func flattenDiskListFilterCloudInit(d *schema.ResourceData, disks []*v3.VMDisk)
 					}
 				}
 				if !match {
-					potentialCloudInitIDs = append(potentialCloudInitIDs, *eDisk.UUID)
+					potentialCloudInitHelpers = append(potentialCloudInitHelpers, CloudInitHelper{
+						UUID:           *eDisk.UUID,
+						CloudInitCDrom: eDisk,
+						Index:          index,
+					})
+					if !removeCloudInit {
+						filteredDiskList = append(filteredDiskList, eDisk)
+					}
 				}
 			}
 		}
-		if len(potentialCloudInitIDs) == 1 {
-			cloudInitCdromUUID = potentialCloudInitIDs[0]
-			d.Set("cloud_init_cdrom_uuid", cloudInitCdromUUID)
+		if len(potentialCloudInitHelpers) == 1 {
+			cloudInitHelper = &potentialCloudInitHelpers[0]
+			cloudInitCdromUUID = cloudInitHelper.UUID
+			// d.Set("cloud_init_cdrom_uuid", cloudInitCdromUUID)
```

Comment:
is it necessary to leave it as a comment?

Reply:
Removed.